### ISSUE TITLE
t1404 fails

### ISSUE THREAD

**Issue by dscho** [git-for-windows/git#185]
t1404 fails because the expected error message

```
''refs/1l/c' exists; cannot create 'refs/1l/c/x' output.
```

does not match the actual error message

```
fatal: Cannot lock ref 'refs/1l/c/x': unable to create directory for .git/refs/1l/c/x
```

My hunch is that this has to do with 194c3f14966dc91d3ef976c386cd411636452c8d but I have no evidence for that yet, and will only be able to come back to this issue later today.

**Comment by dscho** [comment]
Update: our `mingw_lstat()` does not set `errno = ENOTDIR` when appropriate, that is at least one culprit.

**Comment by dscho** [comment]
Fixed in 731fe28
